contents/tutorials/mcp-analytics.mdx

```diff
@@ -0,0 +1,476 @@
+// In src/server.ts
+server.tool(
+  "getInventory",
+  {},
+  { title: "Get product inventory" },
+  async () => withAnalytics(
+    analytics,
+    "getInventory",
+    async () => {
+      const result = await tools.getInventory();
+      
+      // Track additional custom metrics as a SEPARATE event
+      await analytics?.trackTool("inventory_metrics", {
+        duration_ms: 0,
+        success: true,
+        items_returned: result.length,
+        cache_hit: false
+      });
+      
+      return result;
+    }
+  )
+);
+```
+
+## 5. Implement PostHog tracking
+
+Now let's send those analytics somewhere useful. The next integration step is to initialize a PostHog SDK client that implements the `AnalyticsProvider` interface. 
+
+```typescript file="src/posthog.ts"
+// src/posthog.ts
+import { PostHog } from "posthog-node";
+import { AnalyticsProvider } from "./analytics.js";
+
+export class PostHogAnalyticsProvider implements AnalyticsProvider {
+  private client: PostHog;
+  private sessionId: string;
+
+  constructor(apiKey: string, host?: string) {
+    this.client = new PostHog(apiKey, { host });
+    this.sessionId = `mcp_${Date.now()}_${process.pid}`;
+  }
+  
+  async trackTool(toolName: string, result: any): Promise<void> {
+    this.client.capture({
+      distinctId: this.sessionId,
+      event: "mcp_tool_executed",
+      properties: { tool_name: toolName, ...result }
+    });
+    
+    console.error(
+      `[Analytics] ${toolName}: ${result.success ? "✓" : "✗"} (${result.duration_ms}ms)`
+    );
+  }
+  
+  async trackError(error: Error, context: any): Promise<void> {
+    this.client?.captureException(error, undefined, {
+        sessionId: this.sessionId,
+        duration_ms: context.duration_ms,
+        args: this.anonymizeData ? this.anonymize(context.args) : context.args,
+    });
+  }
+  
+  async close(): Promise<void> {
+    await this.client.shutdown();
+  }
+}
+```
+
+The `PostHogAnalyticsProvider` class leverages the PostHog [Node.js SDK](/docs/libraries/node) to capture [custom events](/docs/product-analytics/capture-events) for product analytics and exceptions for built-in [error tracking](/docs/error-tracking).
+
+## 6. Create the MCP server
+
+The last implementation step is to create the MCP server with our `PostHogAnalyticsProvider` instance.
+
+```typescript file="src/index.ts"
+// src/index.ts
+import "dotenv/config";
+import { createServer } from "./server.js";
+import { PostHogAnalyticsProvider } from "./posthog.js";
+
+async function main() {
+  let analytics;
+  
+  if (process.env.POSTHOG_API_KEY) {
+    analytics = new PostHogAnalyticsProvider(
+      process.env.POSTHOG_API_KEY,
+      process.env.POSTHOG_HOST
+    );
+  }
+
+  const server = await createServer(analytics);
+
+  process.on("SIGINT", async () => {
+    await server.close();
+    await analytics?.close();
+    process.exit(0);
+  });
+}
+
+main().catch(console.error);
+```
+
+## 7. Test with Claude Desktop
```

Comment:
I think there is a bit more here to actually get it running. I haven't done this before and I couldn't using these instructions.